python/lsst/ap/association/diaCalculationPlugins.py

```diff
@@ -919,3 +921,134 @@ def fail(self, diaObject, filterName, error=None):
             Error to pass.
         """
         diaObject["{}PSFluxStetsonJ".format(filterName)] = np.nan
+
+
+class WeightedMeanDiaTotFluxConfig(DiaObjectCalculationPluginConfig):
+    pass
+
+
+@register("ap_meanTotFlux")
+class WeightedMeanDiaTotFlux(DiaObjectCalculationPlugin):
+    """Compute the weighted mean and mean error on the point source fluxes
+    forced photometered at the DiaSource location in the calibrated image.
+
+    Additionally store number of usable data points.
+    """
+
+    ConfigClass = WeightedMeanDiaPsFluxConfig
+    outputCols = ["TOTFluxMean", "TOTFluxMeanErr"]
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_CATALOGCALCULATION
+
+    def calculate(self,
+                  diaObject,
+                  diaSources,
+                  filterDiaFluxes,
+                  filterName,
+                  **kwargs):
+        """Compute the weighted mean and mean error of the point source flux.
+
+        Parameters
+        ----------
+        diaObject : `dict`
+            Summary object to store values in.
+        diaSources : `pandas.DataFrame`
+            DataFrame representing all diaSources associated with this
+            diaObject.
+        filterDiaFluxes : `pandas.DataFrame`
+            DataFrame representing diaSources associated with this
+            diaObject that are observed in the band pass ``filterName``.
+        filterName : `str`
+            Simple, string name of the filter for the flux being calculated.
```

Comment:
What does simple mean here?

Reply:
Simple as in the string name like "g" or "r". Usually they are single characters but I don't think they have to be.